spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisStreamInboundChannelAdapter.java

```diff
@@ -0,0 +1,272 @@
+import org.springframework.data.redis.hash.ObjectHashMapper;
+import org.springframework.data.redis.serializer.JdkSerializationRedisSerializer;
+import org.springframework.data.redis.serializer.RedisSerializer;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+import org.springframework.data.redis.stream.StreamListener;
+import org.springframework.data.redis.stream.StreamMessageListenerContainer;
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.converter.MessageConverter;
+import org.springframework.messaging.converter.SimpleMessageConverter;
+import org.springframework.util.Assert;
+import org.springframework.util.ErrorHandler;
+import org.springframework.util.StringUtils;
+
+/**
+ * Read Message from a Redis Stream and publish it to the indicated output channel.
+ */
+public class RedisStreamInboundChannelAdapter extends MessageProducerSupport {
+
+    private volatile StreamMessageListenerContainer container;
+
+    private final String                             streamName;
+
+    private volatile MessageListenerContainerOptions containerOptions;
+
+    private volatile @Nullable	String                consumerGroupName;
+
+    private volatile @Nullable	String                consumerName;
+
+    private volatile boolean                         createGroupeIfNotExist = false;
+
+    private volatile ReadOffset readOffset             = ReadOffset.lastConsumed();
+
+    private final RedisConnectionFactory connectionFactory;
+
+    private final RedisTemplate<String, ?> template;
+
+    private final MessageConverter messageConverter       = new SimpleMessageConverter();
+
+    public RedisStreamInboundChannelAdapter( String streamName, RedisConnectionFactory connectionFactory ) {
+        this.streamName = streamName;
+        this.connectionFactory = connectionFactory;
+        template = new RedisTemplate<>();
+        template.setConnectionFactory( connectionFactory );
+        template.setKeySerializer( new StringRedisSerializer() );
+        template.afterPropertiesSet();
+    }
+
+    private void setContainer( StreamMessageListenerContainer container ) {
+        this.container = container;
+    }
+
+    @Override
+    protected void onInit() {
+        super.onInit();
+        this.createGroup();
+        this.listenMessage();
+    }
+
+    @Override
+    protected void doStart() {
+        super.doStart();
+        this.container.start();
+    }
+
+    @Override
+    protected void doStop() {
+        super.doStop();
+        this.container.stop();
+    }
+
+    @Override
+    public String getComponentType() {
+        return "redis:stream-inbound-channel-adapter";
+    }
+
+    private Message<?> convertMessage( Record<Object, Object> message ) {
+        return this.messageConverter.toMessage( Record.of( message ), null );
+    }
```

Comment:
@artembilan How to improve this converter? 
Currently I send back a `Record` to the output channel, it would be ideal to extract the paylod and headers and construct a real `Message`  before publishing it.